dask/distributed/worker.py

```diff
@@ -371,6 +375,8 @@ def block(self):
         """
         self._listen_workers_thread.join()
         self._listen_scheduler_thread.join()
+        self._heartbeat_thread.event.set()  # stop heartbeat
```

Comment:
Is there another way these threads are closed? Sometimes py.test does not return my prompt for ~5 seconds, the amount of time of a heartbeat.